### PR TITLE
gh-684 document box.snapshot() for vinyl

### PR DESCRIPTION
**Pull request by Onvember**

### CODE REVIEW

File: doc/reference/reference_lua/box_snapshot.rst [hunk 1]
```diff
@@ -74,3 +76,22 @@
     in automation: a signal provides no way to find out whether the snapshot
     was taken successfully or not.
 
+    **Vinyl**
+
+    In vinyl, inserted data is stacked in memory until the limit, set in
+    :ref:`vinyl_memory <cfg_storage-vinyl_memory>` parameter, is reached. Then
+    vinyl automatically dumps it to a disc. ``box.snapshot()`` forces doing
```
Comment (lenkis) [inline doc/reference/reference_lua/box_snapshot.rst:83]:
```suggestion
    vinyl automatically dumps it to the disc. ``box.snapshot()`` forces
```

File: doc/reference/reference_lua/box_snapshot.rst [hunk 2]
```diff
@@ -74,3 +76,22 @@
...
+    this dump in order to have the ability to recover from this checkpoint.
+    The snapshot files are stored in :samp:`{space_id}/{index_id}/*.run`.
+    Thus, strictly all the data that was written at the time of LSN of the
+    checkpoint is in the `*.run` files on the disk, and all operations that happened
+    after the checkpoint will be written in the `*.xlog`. All dump files created
+    by ``box.snapshot()`` are consistent and have the same LSN as checkpoint.
+
+    At the checkpoint vinyl also rotates the metadata log `*.vylog`, containing
+    data manipulation operations like «create file» and «delete file». It goes
```
Comment (lenkis) [inline doc/reference/reference_lua/box_snapshot.rst:92]:
```suggestion
    data manipulation operations like "create file" and "delete file". It goes
```

File: doc/reference/reference_lua/box_snapshot.rst [hunk 3]
```diff
@@ -74,3 +76,22 @@
...
+    through the log, removes duplicating operations from the memory and creates
+    new `*.vylog` file, giving it the name according to the
```
Comment (lenkis) [inline doc/reference/reference_lua/box_snapshot.rst:94]:
```suggestion
    a new `*.vylog` file, giving it the name according to the
```

File: doc/reference/reference_lua/box_snapshot.rst [hunk 4]
```diff
@@ -74,3 +76,22 @@
...
+    :ref:`vclock <box_introspection-box_info>` of the new checkpoint, with
+    «create» operations only. This procedure cleans `*.vylog` and is useful for
```
Comment (lenkis) [inline doc/reference/reference_lua/box_snapshot.rst:96]:
```suggestion
    "create" operations only. This procedure cleans `*.vylog` and is useful for
```

**Review by lenkis — CHANGES_REQUESTED** [review]

File: doc/reference/reference_lua/box_snapshot.rst [hunk 5]
```diff
@@ -74,3 +76,22 @@
     in automation: a signal provides no way to find out whether the snapshot
     was taken successfully or not.
 
+    **Vinyl**
+
+    In vinyl, inserted data is stacked in memory until the limit, set in
+    :ref:`vinyl_memory <cfg_storage-vinyl_memory>` parameter, is reached. Then
```
Comment (lenkis) [inline doc/reference/reference_lua/box_snapshot.rst:82]:
```suggestion
    the :ref:`vinyl_memory <cfg_storage-vinyl_memory>` parameter, is reached. Then
```

File: doc/reference/reference_lua/box_snapshot.rst [hunk 6]
```diff
@@ -74,3 +76,22 @@
...
+    vinyl automatically dumps it to the disc. ``box.snapshot()`` forces
+    this dump in order to have the ability to recover from this checkpoint.
+    The snapshot files are stored in :samp:`{space_id}/{index_id}/*.run`.
+    Thus, strictly all the data that was written at the time of LSN of the
+    checkpoint is in the `*.run` files on the disk, and all operations that happened
```
Comment (lenkis) [inline doc/reference/reference_lua/box_snapshot.rst:87]: Please mark all file names in this commit with double quotes:

```
 ``*.run`` but not `*.run`
```

Single quotes will be rendered in italics, which is harder to read than the monotype font that we get with double quotes.

**Review by lenkis — CHANGES_REQUESTED** [review]

File: doc/reference/reference_lua/box_snapshot.rst (hunk 5 above)
Comment (Onvember) [reply doc/reference/reference_lua/box_snapshot.rst:82]: fixed in 36c09b2

File: doc/reference/reference_lua/box_snapshot.rst (hunk 6 above)
Comment (Onvember) [reply doc/reference/reference_lua/box_snapshot.rst]: fixed in 36c09b2

**Pull request merged**